### PR TITLE
Change the writemode to overwrite

### PR DESCRIPTION
**Pull request by nrc**
r? @topecongiro

### CODE REVIEW

**Comment by nrc** [comment]
Closes #873

File: README.md [hunk 1]
```diff
@@ -90,10 +90,10 @@ read data from stdin. Alternatively, you can use `cargo fmt` to format all
 binary and library targets of your crate.
 
 You'll probably want to specify the write mode. Currently, there are modes for
-diff, replace, overwrite, display, coverage, and checkstyle.
+`diff`, `replace`, `overwrite`, `display`, `coverage`, and `checkstyle`.
```
Comment (topecongiro) [inline README.md]: Looks like `Plain` mode has been missing here and elsewhere

File: src/bin/rustfmt.rs [hunk 2]
```diff
@@ -120,16 +115,10 @@ fn make_opts() -> Options {
...
-        "w",
-        "no-warn-write-mode",
-        "inhibit warning about write-mode change",
-    );
     opts.optopt(
         "",
         "write-mode",
-        "mode to write in (not usable when piping from stdin)",
+        "how to write output (not usable when piping from stdin)",
         "[replace|overwrite|display|diff|coverage|checkstyle]",
```
Comment (topecongiro) [inline src/bin/rustfmt.rs]: and here

File: src/config.rs [hunk 3]
```diff
@@ -596,7 +596,7 @@ create_config! {
     spaces_within_square_brackets: bool, false, "Put spaces within non-empty square brackets";
     spaces_within_parens: bool, false, "Put spaces within non-empty parentheses";
     use_try_shorthand: bool, false, "Replace uses of the try! macro by the ? shorthand";
-    write_mode: WriteMode, WriteMode::Replace,
+    write_mode: WriteMode, WriteMode::Overwrite,
         "What Write Mode to use when none is supplied: Replace, Overwrite, Display, Diff, Coverage";
```
Comment (topecongiro) [inline src/config.rs]: and here

File: README.md [hunk 4]
```diff
@@ -90,10 +90,10 @@ read data from stdin. Alternatively, you can use `cargo fmt` to format all
...
 
-* `replace` Is the default and overwrites the original files after creating backups of the files.
-* `overwrite` Overwrites the original files _without_ creating backups.
+* `overwrite` Is the default and overwrites the original files _without_ creating backups.
```
Comment (topecongiro) [inline README.md]: and here

**Comment by topecongiro** [comment]
All the changes LGTM.
I found that we have been missing `Plain` mode in various places of documents, so could you add it in this PR?

**Comment by nrc** [comment]
Thanks! I added notes about plain mode

**Pull request merged**